apps/vmq_plugin/src/vmq_plugin_mgr.erl

```diff
@@ -514,6 +515,26 @@ check_plugins([{application, App, Options} | Rest], Acc) ->
 check_plugins([], CheckedHooks) ->
     {ok, lists:reverse(CheckedHooks)}.
 
+set_plugin_priority_order(Plugins) ->
+    PriorityMap = application:get_env(vmq_plugin, priority, #{}),
+    lists:sort(
+        fun({_, PluginA, _}, {_, PluginB, _}) ->
+            PriorityA = maps:get(PluginA, PriorityMap, undefined),
+            PriorityB = maps:get(PluginB, PriorityMap, undefined),
+            case {PriorityA, PriorityB} of
+                % No priority defined for either, keep original order
+                {undefined, undefined} -> true;
+                % Keep PluginB before PluginA if PluginB has no priority
```

Comment:
Don't you think any plugin with some priority number should always be executed before plugins without any priority?

Reply:
Yes, The behaviour is that way only.
In this case, the plugin list which we are sorting contains all types of plugins and modules. it can have internal as well as external plugins.
So for internal plugins, we are not mapping any priority in vernemq.conf so it will be undefined.
and for external plugins if we set any priority in vernemq.conf then it will be executed before any external plugin for which the priority is not set and by default, it is infinity so it will be loaded in the end.
commit: 2adbc605bf307aa2d33e76c346a17692db633e25